docs/reference/files/file-charmcraft-yaml.rst

```diff
@@ -0,0 +1,901 @@
+            architectures:
+              - <arch>
+        run-on:
+          - name: <name>
+            channel: <channel>
+            architectures:
+              - <arch>
+
+The ``run-on`` part of each entry is optional and defaults to what's specified in the corresponding ``build-on``. In both structures the list of architecture strings is also optional, defaulting to the architecture of the current machine.
+
+The short form is more concise and simple (at the cost of being less flexible):
+
+.. code::
+
+    bases:
+      - name: <name>
+        channel: <channel>
+        architectures:
+          - <arch>
+
+It implies that the specified base is to be used for both ``build-on`` and ``run-on``. As above, the list of architecture strings is also optional, defaulting to the machine architecture.
+
+
+.. dropdown:: Example
+
+    .. code::
+
+        bases:
+          - build-on:
+              - name: ubuntu
+                channel: '22.04'
+                architectures:
+                  - amd64
+                  - riscv64
+              - name: ubuntu
+                channel: '20.04'
+                architectures:
+                  - amd64
+                  - arm64
+            run-on:
+              - name: ubuntu
+                channel: '22.04'
+                architectures:
+                  - amd64
+              - name: ubuntu
+                channel: '22.04'
+                architectures:
+                  - riscv64
+              - name: ubuntu
+                channel: '22.04'
+                architectures:
+                  - arm64
+          - build-on:
+              - name: ubuntu
+                channel: '24.04'
+            run-on:
+              - name: ubuntu
+                channel: '24.04'
+                architectures:
+                  - amd64
+                  - arm64
+                  - riscv64
+                  - s390x
+                  - ppc64el
+                  - armhf
+
+.. _file-charmcraft-yaml-build-base:
+``build-base``
+==============
+
+**Status:** Only valid if `base`_ is a development base.
+
+**Purpose:** Specifies the operating system on which the charm will be built.
+
+**Structure:**
+
+.. code::
+
+    build-base: <distro>@<series>
+
+**Example:**
+
+.. code::
+
+    base: ubuntu@devel
+
+.. _file-charmcraft-yaml-charm-libs:
+``charm-libs``
+==============
+
+**Status:** Optional.
+
+**Purpose:** Declares charm libraries for Charmcraft to include in the charm project. For each lib, include both the lib name (in ``<charm>.<library>`` format) and the lib version (in ``"<api version>[.<patch version>]"`` string format).
+
+**Structure:**
+
+.. code::
+
+    charm-libs:
+      - lib: <charm>.<library>
+        version: "<api>[.<patch>]"  # Must be a string, not a number.
+
+**Example:**
+
+.. literalinclude:: charmcraft-sample-charm.yaml
+    :start-at: charm-libs:
+    :end-before: config:
+
+.. _file-charmcraft-yaml-charmhub:
+``charmhub``
+=====
+
+
+.. warning::
```

Comment:
Per the [rST style guide](https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/#notes), we use `caution` instead of `warning`.

Reply:
```suggestion
.. caution::
```